R/zchunk_L232.industry_CHINA.R

```diff
@@ -0,0 +1,537 @@
+
+    # get calibrated input of industrial energy use technologies, including cogen
+    L132.in_EJ_province_indnochp_F %>%
+      bind_rows(L132.in_EJ_province_indchp_F) %>%
+      complete(nesting(province, sector, fuel), year = c(year, MODEL_BASE_YEARS)) %>%
+      group_by(province, sector, fuel) %>% mutate(value = approx_fun(year, value)) %>%
+      ungroup %>% filter(year %in% MODEL_BASE_YEARS) %>%
+      rename(region = province) %>%
+      left_join_keep_first_only(calibrated_techs, by = c("sector", "fuel")) %>%
+      rename(stub.technology = technology) ->
+      L232.in_EJ_province_indenergy_F_Yh
+    L232.in_EJ_province_indenergy_F_Yh %>% select(LEVEL2_DATA_NAMES[["StubTechYr"]], "value") %>%
+      left_join_keep_first_only(select(A32.globaltech_eff, subsector, technology, minicam.energy.input),
+                                by = c("subsector", "stub.technology" = "technology")) %>%
+      mutate(calibrated.value = round(value, energy.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             tech.share.weight = if_else(calibrated.value > 0, 1, 0)) %>%
+      group_by(region, supplysector, subsector, year) %>%
+      mutate(x = sum(calibrated.value),
+             subs.share.weight = if_else(x > 0, 1, 0)) %>% ungroup %>%
+      # ^^ sets up variable (x) for defining subsector shareweight
+      select(LEVEL2_DATA_NAMES[["StubTechCalInput"]]) ->
+      L232.StubTechCalInput_indenergy_CHINA  ## OUTPUT
+
+    # get calibrated input of industrial feedstock technologies
+    L132.in_EJ_province_indfeed_F %>%
+      complete(nesting(province, sector, fuel), year = c(year, MODEL_BASE_YEARS)) %>%
+      group_by(province, sector, fuel) %>% mutate(value = approx_fun(year, value)) %>%
+      ungroup %>% filter(year %in% MODEL_BASE_YEARS) %>%
+      rename(region = province) %>%
+      left_join_keep_first_only(calibrated_techs, by = c("sector", "fuel")) %>%
+      select(-calibration, -secondary.output) %>%
+      rename(stub.technology = technology) ->
+      L232.in_EJ_province_indfeed_F_Yh
+    L232.in_EJ_province_indfeed_F_Yh %>% select(LEVEL2_DATA_NAMES[["StubTechYr"]], "value") %>%
+      left_join_keep_first_only(select(A32.globaltech_eff, subsector, technology, minicam.energy.input),
+                                by = c("subsector", "stub.technology" = "technology")) %>%
+      mutate(calibrated.value = round(value, energy.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             tech.share.weight = if_else(calibrated.value > 0, 1, 0)) %>%
+      group_by(region, supplysector, subsector, year) %>% mutate(x = sum(calibrated.value)) %>%
+      # ^^ sets up variable (x) for defining subsector shareweight
+      mutate(subs.share.weight = if_else(x > 0, 1, 0)) %>% ungroup %>%
+      select(LEVEL2_DATA_NAMES[["StubTechCalInput"]]) ->
+      L232.StubTechCalInput_indfeed_CHINA  ## OUTPUT
+
+    # get industrial sector calibrated output
+    A32.globaltech_eff %>%
```

Comment:
add a comment, "First, need to interpolate industrial tech efficiencies" or something along those lines.